libdrgn/python/thread.c

```diff
@@ -56,6 +56,21 @@ static DrgnObject *Thread_get_object(Thread *self)
 	return_ptr(ret);
 }
 
+static PyObject *Thread_get_name(Thread *self)
+{
+	_cleanup_free_ char *ret = NULL;
+	struct drgn_error *err = drgn_thread_name(&self->thread, &ret);
+	if (err)
+		return set_drgn_error(err);
+	if (!ret) {
+		return Py_None;
+	}
+	_cleanup_pydecref_ PyObject *buf = PyBytes_FromString(ret);
+	if (!buf)
+		return NULL;
+	return_ptr(buf);
```

Comment:
I'm on the fence about this, so let me know if you disagree, but I think making this a `str` instead of `bytes` is a nicer interface. We just need to be carefuly about decoding errors. The filesystem decoder should be good enough. Also, no need for the intermediate variable here.
```suggestion
	return PyUnicode_DecodeFSDefault(name);
```

Reply:
Agree, I kinda like str better too, I just wanted to avoid encoding issues. But if there's a convenient function for decoding based on default encoder, lets use it!